src/index.js

```diff
@@ -607,6 +607,95 @@ export default function (babel) {
     path.unshiftContainer("body", declarations);
   }
 
+  function generateForInIterator (path, type: "array" | "object") {
+    let idx = path.node.idx || path.scope.generateUidIdentifier("i");
+    let keys = path.scope.generateUidIdentifier("keys");
```

Comment:
I went with just assigning it further down where it's needed - let me know if that works for you.